### DISCUSSION TITLE
How do I handle long values in mongo db?

### DISCUSSION THREAD

**Discussion by bwpark** [mikro-orm/mikro-orm#3625] [Q&A]
When reading a long type value from mongo db, it is sometimes displayed as a number or as a json object like the one below.
How do I get number from rest api?

"id": {  "low": 1157459400, "high": 3792289, "unsigned": false },

**Comment by B4nan** [comment]
> How do I get number from rest api?

You can't, API response will be JSON, and there you cannot have anything else than a JS number, which is a 64 bit float. There is no way to have a long number in JSON, only as a string.

(long number won't fit into JS float, not in its full range of allowed values)

**Reply by bwpark** [reply]
When I call the find function
const issues = await this.repository.find(where)

the issues[0].id value can be an object like {id: {low: 1940474783, high: 2860538, unsigned: false} , or it can be created with a number like issues[1].id  = 4430809979435385.

Don't you think this is a serious bug?

**Reply by B4nan** [reply]
No I dont, you clearly did not understand what I just told you. You cannot transfer a long number in a JSON (= your API response) as a number, that is simply not possible - the JSON standard (nor JS) support that. If you want to transfer a long number (without loosing the actual value), you need to use a string.

The number you mentioned would fit into JS number, but if you add one more digit, it won't, try to print `44308099794353999` in the JS console, you will see a different number (44308099794354000) being printed, as JS can't represent the input one.

```
console.log(44308099794353999); // prints 44308099794354000
```

You can do anything you want with the value, with [custom mapped types](https://mikro-orm.io/docs/custom-types) you can even convert it to a JS number on load. But once again - you can lose precision this way, it can't be the default, and in most cases this is wrong (why do you want to use long number if you dont care about working the actual value precisely?).
